sources/web/datalab/polymer/modules/template-manager/template-manager.ts

```diff
@@ -44,7 +44,20 @@ class NotebookTemplate {
 
   constructor(fileId: DatalabFileId, parameters: TemplateParameter[]) {
     this.fileId = fileId;
-    this.parameters = parameters;
+    this.parameters = NotebookTemplate._sanitizeParams(parameters);
+  }
+
+  /**
+   * Strictly sanitizes the parameters.
+   */
+  private static _sanitizeParams(params: TemplateParameter[]) {
+    params.forEach((p: TemplateParameter, i) => {
```

Comment:
Isn't the notebook name part of the parameters? And that can be anything the user wants?

Reply:
The filename isn't part of these params.  We currently allow anything for the name and sanitize those separately when displaying them, since we don't control all of the paths for creating a file (i.e. a user could create one named with a script tag themselves outside of Datalab)

Reply:
Isn't the file name going to be used to populate a markdown cell within the template to serve as the starting title?

Reply:
We are not yet doing that for any of our templates.  To do so, we'd have to pass the fileName as a parameter to the template manager, in which case it would go through this sanitization.

Reply:
That _was_ the plan for the templates -- first cell contains the title based on the name.